lua/starfall/libs_cl/render.lua

```diff
@@ -868,6 +887,22 @@ function render_library.setRGBA(r, g, b, a)
 	surface.SetTextColor(r, g, b, a)
 end
 
+--- Gets the drawing tint. Internally, calls render.getColorModulation and render.getBlend, multiplies the values by 255, then returns a color object.
+-- @return Color The current color & blend modulation as a color
+function render_library.getTint()
+	local r, g, b = render.GetColorModulation()
+	local a = render.GetBlend()
+		
+	return cwrap{r * 255, g * 255, b * 255, a * 255}
```

Comment:
This actually needs to use `r`, `g`, `b`, `a` keys, because that's how GLua Color stores the components. My bad.

Could make it faster by doing `setmetatable({ r * 255, g * 255, b * 255, a * 255 }, col_meta)`. Starfall already does it like that in few places: `vec_methods:getAngle()`, `math_library.bezierVector`, `math_library.bezierVectorCubic`, `vwrap2` in `vmatrix.lua`.

Best to ask Sparky if it's OK to do this or not.

Reply:
@thegrb93

Reply:
This code is correct

Reply:
> This code is correct

It will error, because `cwrap` will try to construct a Color object using `r`, `g`, `b`, `a` keys.

Reply:
> > This code is correct
> 
> It will error, because `cwrap` will try to construct a Color object using `r`, `g`, `b`, `a` keys.

My apologies, I thought I switched it earlier. I added the changes just now.